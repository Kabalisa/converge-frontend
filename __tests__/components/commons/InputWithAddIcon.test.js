import React from 'react';
import { shallow } from 'enzyme';
import SingleInputWithAddIcon from '../../../src/components/commons/InputAddIcon/InputWithPlusIcon';

describe.only('Input with plus icon component', () => {
  const props = {
    inputPlaceholder: 'Building',
    pressed: jest.fn(),
    name: 'building_1',
    onTextChange: jest.fn(),
    values: {},
    errors: {},
  };
  const wrapper = shallow(<SingleInputWithAddIcon {...props} />);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should receive inputPlaceHolderProps', () => {
    expect(wrapper.props().children[0].props.placeholder).toBe(props.inputPlaceholder);
  });

  it('should call pressed when click plus icon', () => {
    wrapper.props().children[0].props.icon.props.onClick();
    expect(props.pressed).toHaveBeenCalled();
  });

  it('Should call onTextChange when user start typing in the text input', () => {
    wrapper.props().children[0].props.onChange();
    expect(props.onTextChange).toHaveBeenCalled();
  });

  it('Should call onTextChange When input lose focus', () => {
    wrapper.props().children[0].props.onBlur();
    expect(props.onTextChange).toHaveBeenCalled();
  });

  it('Should receive value from props', () => {
    props.values = {
      building_1: 'WestAf',
    };
    const component = shallow(<SingleInputWithAddIcon {...props} />);
    expect(component.props().children[0].props.value).toBe('WestAf');
  });

  it('Should receive errors from props', () => {
    props.errors = {
      building_1: '"the building name" should be at least 2 characters in length',
    };

    const component = shallow(<SingleInputWithAddIcon {...props} />);
    expect(component.props().children[0].props.error).toBeTruthy();
    expect(component.find('.building-error-area-style').text()).toBe('"the building name" should be at least 2 characters in length');
  });
});
