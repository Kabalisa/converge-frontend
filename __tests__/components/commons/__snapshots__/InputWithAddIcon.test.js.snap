// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input with plus icon component should render properly 1`] = `
<div
  className="InputWithAddIcon-container"
>
  <Input
    className="InputWithAddIcon-style"
    error={false}
    icon={
      <Icon
        as="i"
        className="plus-icon"
        color="green"
        disabled={false}
        link={true}
        name="plus circle"
        onClick={[Function]}
        size="large"
      />
    }
    name="building_1"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder="Building"
    size="large"
    type="text"
    value=""
  />
</div>
`;
